ui/src/components/dialog/__tests__/QDialog.cy.js

```diff
@@ -1,21 +1,131 @@
+
+function mountQDialogWrapper (options) {
+  return cy.mount(DialogWrapper, options)
+}
+
+function getHostElement () {
+  return cy.dataCy('dialog-card')
+}
+
+function closeDialogViaBackdrop () {
+  return cy.get('.q-dialog__backdrop').click({ force: true })
+}
+
+// TODO: this only works in `withinDialog` context,
+// as when using it outside it `.root()` will yeld "html" tag
+// and Cypress won't be able to find anything with `.closest('body')`
+// since it only searches upwards
+function closeDialogViaEscKey () {
+  // Official way to escape `within` context
+  // https://docs.cypress.io/api/commands/within#Temporarily-escape
+  return cy.root().closest('body').type('{esc}')
+}
+
+// TODO: we'll add this check to the AE helper to automatically test persistent dialogs
+// TODO: take into consideration that seamless dialogs have always 'aria-modal' false
+// TODO: take into consideration that dialogs without backdrop have always 'aria-modal' false
+// Check if there's a more reliable way to check for persistent dialogs
+function assertPersistentDialogExists () {
+  cy.dataCy('dialog-form').should('not.have.attr', 'aria-modal', 'false')
+}
+
 describe('Dialog API', () => {
   describe('Props', () => {
     describe('Category: behavior', () => {
       describe('(prop): persistent', () => {
-        it.skip(' ', () => {
-          //
+        it('should display a persistent dialog', () => {
+          const model = ref(true)
+          mountQDialogWrapper({
+            props: {
+              ...vModelAdapter(model),
+              persistent: true
+            }
+          })
+
+          cy.withinDialog({
+            persistent: true,
+            fn: () => {
+              closeDialogViaBackdrop()
+            }
+          })
+
+          assertPersistentDialogExists()
+
+          cy.withinDialog({
+            persistent: true,
+            fn: () => {
+              closeDialogViaEscKey()
+            }
+          })
+
+          assertPersistentDialogExists()
         })
       })
 
       describe('(prop): no-esc-dismiss', () => {
-        it.skip(' ', () => {
-          //
+        it('should not allow closing the dialog with the escape key', () => {
+          const model = ref(true)
+          mountQDialogWrapper({
+            props: {
+              ...vModelAdapter(model),
+              noEscDismiss: true
+            }
+          })
+
+          cy.withinDialog({
+            persistent: true,
+            fn: () => {
+              closeDialogViaEscKey()
+            }
+          })
+
+          assertPersistentDialogExists()
+
+          cy.wrap()
+            .then(async () => {
+              // Care as you need to wait setProps to complete its execution
+              // before moving on or the tests will fail due to the race condition
+              await Cypress.vueWrapper.setProps({ noEscDismiss: false })
+            })
+
+          cy.withinDialog(() => {
```

Comment:
Try to use `cy.withinDialog` whenever possible, as it automates some checks about Dialog existence and not existence for you